### PR TITLE
Set up "Code Ownership"

### LINKED ISSUE #3312: Set up "Code Ownership"
For those of you who weren't at the meeting from #3302, GitHub supports assigning [code ownership](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners). While it is a bit of a misnomer, it really means that all "owners" for a given subsection of code "are automatically requested for review when someone opens a pull request that modifies [code that they are marked as an owner of]." This essentially just formalizes practices we already follow and allows those who wish to stay involved in Drasil after moving on an easier way to jump in.

I've started a list of pieces of code that certain Drasil contributors might like to be assigned ownership of below; feel free to edit/add to these lists and check off your name once you are happy with the scope of your ownership. (Note that this can always be changed later and can also be empty if you don't wish to be involved in Drasil anymore!) I will then, at some point in the future, create the necessary file and close this issue; even then, ownership can be changed.

- [ ] @AKM11:
- [ ] @Ant13731:
- [ ] @Awurama-N:
- [ ] @balacij:
- [x] @bmaclach: Projectile, GOOL
- [x] @cd155: DblPendulum, `DifferentialModel`
- [ ] @Daisy-C:
- [ ] @Daniel-Genkin:
- [ ] @danscime:
- [ ] @deviprasad135:
- [ ] @elwazana:
- [ ] @halonazhao:
- [ ] @hmadej:
- [x] @JacquesCarette: All of it lol
- [ ] @jeger1917:
- [ ] @lmawarid:
- [ ] @Mornix:
- [ ] @muhammadaliog3:
- [ ] @muralidn:
- [ ] @Nathaniel-Hu:
- [ ] @niazim3:
- [ ] @njericha:
- [ ] @oluowoj: SglPendulum?
- [ ] @palmerst:
- [ ] @peter-michalski:
- [x] @samm82: Projectile, GlassBR, Template, `Drasil.Symbol` (for `unicodeConv`), `Drasil.Sentence.Fold`
- [x] @smiths: stable
- [ ] @Sodaaaa:
- [ ] @szymczdm:
- [ ] @TamkeenFatima:
- [ ] @tingyuw: Any Jupyter notebooks?
- [ ] @vajihehm:

_Sorry for tagging everyone 😬_

### PR DESCRIPTION
**Pull request by samm82**
Closes #3312

I'm pretty sure I did this right, but I also don't know if there is a way to verify this without it being merged into `master`; looking at files in this branch doesn't show their code ownership as the [GitHub page](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) suggests:

image

### CODE REVIEW

**Review by smiths — COMMENTED** [review]
@samm82 you mention in the file that @JacquesCarette has to be added to every rule because of how Code Ownership works.  This isn't a problem though because @JacquesCarette has always reviewed all pull requests.

There is only one file changed here, so my guess is that @JacquesCarette will approve the pull request and then we'll be able to see if it works as we intended.

**Review by JacquesCarette — APPROVED** [review]

File: .github/workflows/CODEOWNERS
Comment (cd155) [inline .github/workflows/CODEOWNERS]: I feel Jason is more eligible for ModelKinds. The DifferentialModel is one of ModelKinds.

Comment (samm82) [reply .github/workflows/CODEOWNERS]: @balacij You cool with that?

**Comment by samm82** [comment]
@smiths What I meant by that is that for Dr. Carette to be added to every PR, he also has to be added to every line in the CODEOWNERS file (since "sub rules" overwrite "super rules" instead of adding to them)

Also, I verified that after this got merged, code ownership was set up correctly! Weird that it doesn't let you "preview these changes" in whatever branch you're in first....

File: .github/workflows/CODEOWNERS
Comment (balacij) [reply .github/workflows/CODEOWNERS]: Yep, of course :smile:

Comment (samm82) [reply .github/workflows/CODEOWNERS]: @balacij Would you want to be added to the entire `code/drasil-theory/lib/Theory/Drasil/` directory or is there a subset you would prefer?

Comment (balacij) [reply .github/workflows/CODEOWNERS]: The single file is fine :smile: We can adjust as we go along.

**Pull request merged**
